### ISSUE TITLE
Body of method appears to be considered type annotation

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript-Sublime-Plugin#127]
As you can see, it thinks `this.writeMessage` is a function definition, and all this causes highlighting to go off the rails.

![bad scope](https://cloud.githubusercontent.com/assets/972891/7039719/8e5f608c-dd7a-11e4-84ff-b2cd91cef0cc.png)

**Comment by steveluc** [comment]
@DanielRosenwasser can you attach the file or copy into a comment a portion large enough to make this happen?

**Comment by DanielRosenwasser** [comment]
Try this repro:

```TypeScript
class C {
    private f<T extends C>(command: C, args?: string): T {
        var request = { };

        this.foo()
    }

    private g<T extends String>(command: Foo): T {

    }
}
```

**Comment by sccolbert** [comment]
The `method-declaration` rule doesn't handle type parameters, which in general causes highlighting to fail on generic methods.

https://github.com/Microsoft/TypeScript-Sublime-Plugin/blob/master/TypeScript.YAML-tmLanguage#L125

**Comment by steveluc** [comment]
Thanks @sccolbert and @DanielRosenwasser. 

Fixed in commit c0f47cc.
